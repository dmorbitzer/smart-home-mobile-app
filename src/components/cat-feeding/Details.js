import { Text, Box, Divider } from "@react-native-material/core";
import { useState } from "react";

import CatFeedingManual from "./CatFeedingManual";
import DetailsWeekOverview from "./DetailsWeekOverview";
import FeedingTimeHeader from "./FeedingTimeHeader";
import FeedingTimeList from "./FeedingTimeList";
import DeleteFeedingTimeModal from "./cat-feeding-modals/DeleteFeedingTimeModal";

export default function Details(props) {
  const [showModal, setShowModal] = useState(false);
  const [toDeleteFeedingTimeId, setToDeleteFeedingTimeId] = useState(null);

  const openModal = () => {
    setShowModal(true);
  };
  const closeModalHandler = () => {
    setShowModal(false);
  };
  const feedingTimeLongClick = (feedingTimeId) => {
    setToDeleteFeedingTimeId(feedingTimeId);
    openModal();
  };
  return (
    <>
      <Box style={{ alignSelf: "center" }}>
        <Text variant="h1">{props.catName}</Text>
      </Box>
      <Divider style={{ margin: 20 }} />
      <CatFeedingManual
        onFeedNowClick={props.onFeedNowClick}
        feedingTimesData={props.feedingTimesData}
        foodData={props.foodData}
        navigation={props.navigation}
        catId={props.catId}
      />
      <Divider style={{ margin: 20 }} />
      <DetailsWeekOverview
        weekDay={props.weekDay}
        setWeekDay={props.setWeekDay}
      />
      <Divider style={{ margin: 20 }} />
      <FeedingTimeHeader
        navigation={props.navigation}
        weekDay={props.weekDay}
        catId={props.catId}
        catName={props.catName}
        foodData={props.foodData}
      />
      <Divider style={{ margin: 20 }} />
      <FeedingTimeList
        feedingTimeLongClick={feedingTimeLongClick}
        weekDay={props.weekDay}
        feedingTimesData={props.feedingTimesData}
        catName={props.catName}
        catId={props.catId}
      />
      <DeleteFeedingTimeModal
        deleteFeedingTime={props.deleteFeedingTime}
        feedingTimeId={toDeleteFeedingTimeId}
        showModal={showModal}
        closeModalHandler={closeModalHandler}
      />
    </>
  );
}
