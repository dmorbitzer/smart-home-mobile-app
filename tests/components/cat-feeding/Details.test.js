import React from "react";
import renderer from "react-test-renderer";

import Details from "../../../src/components/cat-feeding/Details";
jest.useFakeTimers();
const navigation = null;
const localFoodData = {
  data: {
    food: {
      edges: [
        {
          node: {
            id: "/api/food/7",
            name: "Trockenfutter",
          },
        },
        {
          node: {
            id: "/api/food/8",
            name: "Diät Trockenfutter",
          },
        },
      ],
    },
  },
};
const localFeedingTimesData = {
  data: {
    feedingTimes: {
      edges: [
        {
          node: {
            id: "/api/feeding_times/14",
            cat: {
              id: "/api/cats/246",
              name: "Chris",
            },
            food: {
              name: "Trockenfutter",
            },
            time: "1970-01-01T15:21:38+01:00",
            weekDay: 3,
          },
        },
        {
          node: {
            id: "/api/feeding_times/15",
            cat: {
              id: "/api/cats/247",
              name: "Masha",
            },
            food: {
              name: "Diät Trockenfutter",
            },
            time: "1970-01-01T15:23:02+01:00",
            weekDay: 2,
          },
        },
      ],
    },
  },
};
const onFeedNowClick = (catId, foodId) => {
  console.log(catId, foodId);
};
test("renders cat-feeding Details correctly", () => {
  const tree = renderer
    .create(
      <Details
        foodData={localFoodData.data.food.edges}
        feedintTimesData={localFeedingTimesData.data.feedingTimes.edges[0].node}
        onFeedNowClick={onFeedNowClick}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
