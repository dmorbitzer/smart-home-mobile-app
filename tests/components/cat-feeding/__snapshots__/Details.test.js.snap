// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cat-feeding details correctly 1`] = `
[
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        {
          "alignSelf": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 96,
            "fontWeight": "300",
            "letterSpacing": -1.5,
          },
          {
            "color": "#000000",
          },
          undefined,
        ]
      }
    />
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#00000014",
          "height": 1,
          "marginEnd": undefined,
          "marginStart": undefined,
        },
        {
          "margin": 20,
        },
      ]
    }
  />,
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": undefined,
          "marginBottom": 1,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": 20,
          "marginRight": 20,
          "marginStart": undefined,
          "marginTop": 1,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      onTouchStart={[Function]}
      style={
        {
          "backgroundColor": "lightblue",
          "borderBottomEndRadius": 4,
          "borderBottomStartRadius": 4,
          "borderTopEndRadius": 4,
          "borderTopStartRadius": 4,
          "justifyContent": "center",
          "minHeight": 60,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 1.41,
        }
      }
    >
      <View
        style={
          [
            {
              "borderBottomEndRadius": 4,
              "borderBottomStartRadius": 4,
              "borderTopEndRadius": 4,
              "borderTopStartRadius": 4,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onLayout={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 36,
                "justifyContent": "center",
                "minWidth": 64,
                "paddingEnd": 16,
                "paddingStart": 12,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "marginEnd": 8,
                  "opacity": 1,
                },
                undefined,
              ]
            }
          >
            <Text />
          </View>
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 1.25,
                  "textTransform": "uppercase",
                },
                {
                  "color": "#000000",
                },
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "opacity": 1,
                    "textTransform": "uppercase",
                  },
                  {
                    "fontSize": 25,
                  },
                ],
              ]
            }
          >
            Jetzt f√ºttern
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 0,
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    />
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#00000014",
          "height": 1,
          "marginEnd": undefined,
          "marginStart": undefined,
        },
        {
          "margin": 20,
        },
      ]
    }
  />,
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        [
          {
            "alignContent": undefined,
            "alignItems": "center",
            "alignSelf": undefined,
            "flex": 1,
            "flexBasis": undefined,
            "flexDirection": "row",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": undefined,
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
  >
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              {
                "marginEnd": 3,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              null,
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#00000014",
          "height": 1,
          "marginEnd": undefined,
          "marginStart": undefined,
        },
        {
          "margin": 20,
        },
      ]
    }
  />,
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        [
          {
            "alignContent": undefined,
            "alignItems": undefined,
            "alignSelf": undefined,
            "flex": undefined,
            "flexBasis": undefined,
            "flexDirection": "row",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": undefined,
            "justifyContent": undefined,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 24,
          },
          {
            "color": "#000000",
          },
          [
            {
              "marginLeft": 20,
              "marginTop": 11,
            },
            [
              {
                "marginEnd": 0,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      F√ºtterungszeiten
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": undefined,
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": undefined,
            "borderColor": undefined,
            "borderEndColor": undefined,
            "borderEndWidth": undefined,
            "borderLeftColor": undefined,
            "borderLeftWidth": undefined,
            "borderRadius": undefined,
            "borderRightColor": undefined,
            "borderRightWidth": undefined,
            "borderStartColor": undefined,
            "borderStartWidth": undefined,
            "borderStyle": undefined,
            "borderTopColor": undefined,
            "borderTopEndRadius": undefined,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "borderTopStartRadius": undefined,
            "borderTopWidth": undefined,
            "borderWidth": undefined,
            "bottom": undefined,
            "end": undefined,
            "height": undefined,
            "left": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginEnd": undefined,
            "marginHorizontal": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginStart": undefined,
            "marginTop": undefined,
            "marginVertical": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "opacity": undefined,
            "overflow": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingEnd": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingStart": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "start": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          },
          [
            {
              "alignContent": undefined,
              "alignItems": undefined,
              "alignSelf": "stretch",
              "flex": 1,
              "flexBasis": undefined,
              "flexDirection": undefined,
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": undefined,
              "justifyContent": undefined,
            },
            [
              undefined,
              [
                {
                  "marginEnd": 0,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
    />
    <View
      onTouchStart={[Function]}
      style={
        [
          {
            "borderRadius": 24,
            "height": 48,
            "overflow": "hidden",
            "width": 48,
          },
          [
            undefined,
            [
              null,
              undefined,
            ],
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text />
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#00000014",
          "height": 1,
          "marginEnd": undefined,
          "marginStart": undefined,
        },
        {
          "margin": 20,
        },
      ]
    }
  />,
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "alignSelf": "center",
          "marginTop": 15,
        },
      ]
    }
  >
    Bitte zuerst einen Wochentag w√§hlen!
  </Text>,
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    />
  </View>,
]
`;
